src/compiler/checker.ts

```diff
@@ -28347,7 +28347,15 @@ namespace ts {
                     }
                 case SyntaxKind.CommaToken:
                     if (!compilerOptions.allowUnreachableCode && isSideEffectFree(left) && !isEvalNode(right)) {
-                        error(left, Diagnostics.Left_side_of_comma_operator_is_unused_and_has_no_side_effects);
+                        const sf = getSourceFileOfNode(left);
+                        const isInDiag2657 = sf.parseDiagnostics.some(diag => {

```

Comment:
You could consider using `forEach` or a for/of loop to terminate once `diag.start > left.end`.

Reply:
I didn't catch that, Array.some can quite early, I think it is same as for/of + break

Reply:
I mean that with `.some`, if you don’t find the diagnostic you’re looking for, you’ll keep looking all the way to the end of the array. But since these diagnostics are added in the order that the parser sees them, you know that after you move all the way past the end of `left`, you’re not going to find what you’re looking for at all. So while `.some` can terminate early in the positive case, you also have an opportunity to terminate early in the negative case.

That said, I don’t think it’s a big deal since this is an error scenario and the body of the loop isn’t expensive.